onnxruntime/core/graph/featurizers_ops/featurizers_defs.cc

```diff
@@ -674,6 +678,132 @@ void RegisterLabelEncoderFeaturizerVer1() {
+      //     +-------+-------+-------------------------------------------------+
+      //     | grain | target|        target_lag_2_lead_2_lead_1_lag_1         |
+      //     +=======+=======+=================================================+
+      //     |Walmart| 8     | [[NAN, NAN], [  9,  10], [NAN, NAN], [ 8,   9]] |
+      //     +-------+-------+-------------------------------------------------+
+      //     |Walmart| 9     | [[NAN, NAN], [ 10,  11], [NAN,   8], [ 9,  10]] |
+      //     +-------+-------+-------------------------------------------------+
+      //     |Walmart| 10    | [[NAN,   8], [ 11, NAN], [  8,   9], [10,  11]] |
+      //     +-------+-------+-------------------------------------------------+
+      //     |Walmart| 11    | [[  8,   9], [NAN, NAN], [  9,  10], [11, NAN]] |
+      //     +-------+-------+-------------------------------------------------+
+      //     Basically, if we have an offset of k for the row with row index t,
+      //     target_lag_k[t] = target[t - horizon + k + 1]
+  //)DOC";
+
+  MS_FEATURIZERS_OPERATOR_SCHEMA(LagLeadOperatorTransformer)
+      .SinceVersion(1)
+      .SetDomain(kMSFeaturizersDomain)
+      .Input(
+          0,
+          "State",
+          "State generated during training that is used for prediction",
+          "T0")
+      .Input(
+          1,
+          "Grains",
+          "Grains tensor of shape [R][K].",
+          "GrainT")
+      .Input(
+          2,
+          "Target",
+          "Target tensor of shape [R]",
+          "T")
+      .Output(
+          0,
+          "OutputGrains",
+          "Grains tensor of shape [R][K]",
+          "GrainT")
+      .Output(
+          1,
+          "Output",
+          "Output tensor of shape [R][P][Q]",
+          "T")
+      .TypeConstraint(
+          "T0",
+          {"tensor(uint8)"},
+          "No information is available")
+      .TypeConstraint(
+          "GrainT",
+          {"tensor(string)"},
+          "No information is available")
+      .TypeConstraint(
+          "T",
+          {"tensor(float)", "tensor(double)"},
+          "No information is available")
+      .TypeAndShapeInferenceFunction(
+          [](ONNX_NAMESPACE::InferenceContext& ctx) {
+            propagateElemTypeFromInputToOutput(ctx, 1, 0);
+            auto input_elem_type = ctx.getInputType(2)->tensor_type().elem_type();
+            if (input_elem_type == ONNX_NAMESPACE::TensorProto_DataType_FLOAT) {
+              propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_FLOAT, 1);
+            } else if (input_elem_type == ONNX_NAMESPACE::TensorProto_DataType_DOUBLE) {
+              propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_DOUBLE, 1);
+            } else {
+              fail_type_inference("input 2 is expected to have an accepted type");
```

Comment:
>fail_type_inference("input 2 is expected to have an accepted type"); [](start = 14, length = 68)

The constraints already make sure that the type is acceptable. Just propagate the type.